cockatrice/src/update_checker.cpp

```diff
@@ -37,18 +34,33 @@ void UpdateChecker::check()
 #if defined(Q_OS_OSX)
 bool UpdateChecker::downloadMatchesCurrentOS(QVariant build)
 {
-   return build
+    return build
             .toMap()["name"]
             .toString()
-            .contains("osx");
+            .endsWith(".dmg");
 }
+
 #elif defined(Q_OS_WIN)
 bool UpdateChecker::downloadMatchesCurrentOS(QVariant build)
 {
-       return build
-            .toMap()["name"]
-            .toString()
-            .contains("exe");
+    QString wordSize = QSysInfo::buildAbi().split('-')[2];
+    QString arch;
+    if (wordSize == "llp64") {
+        arch = "win64";
+    } else if (wordSize == "ilp32") {
+        arch = "win32";
+    } else {
+        qWarning() << "Error checking for upgrade version: wordSize is" << wordSize;
+        return false;
+    }
+
+    auto fileName = build
+        .toMap()["name"]
+        .toString();
+    // Checking for .zip is a workaround for the May 6th 2016 release
+    auto zipName = arch + ".zip";
```

Comment:
@ZeldaZach changed one of the two .exe files to zip to default to 64bit downloads on all clients. it was a workaround because otherwise the updater was not working at all and most people are expected to run 64bit.

we could just remove it completely because we have a real differentiation in place now and can offer win32.exe and win64.exe again?!